hal/targets/cmsis/TARGET_ARM_SSG/TARGET_BEETLE/TOOLCHAIN_GCC_ARM/startup_BEETLE.S

```diff
@@ -143,6 +101,15 @@ __isr_vector:
     .globl   Reset_Handler
     .type    Reset_Handler, %function
 Reset_Handler:
+    ldr    r0, =SystemInit
+    blx    r0
+/* The call to uvisor_init() happens independently of uVisor being enabled or
+* not, so it is conditionally compiled only based on FEATURE_UVISOR. */
+#ifdef FEATURE_UVISOR
```

Comment:
Could you change this to `#if defined(FEATURE_UVISOR) && defined(TARGET_UVISOR_SUPPORTED)`?